hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -141,9 +142,13 @@ public QueryResult queryLocalPartitions(String mapName, Predicate predicate, Ite
     }
 
     protected QueryResult tryQueryUsingIndexes(Predicate predicate, Collection<Integer> partitions, MapContainer mapContainer,
-                                               IterationType iterationType) {
+                                               IterationType iterationType, int initialPartitionStateVersion) {
 
-        if (partitionService.hasOnGoingMigrationLocal()) {
+        // if a migration is in progress, do not attempt to use an index as they may have not been created yet.
+        // MapService.getMigrationsInFlight() returns the number of currently executing migrations (for which
+        // beforeMigration has been executed but commit/rollback is not yet executed).
+        // This is a temporary fix for 3.7, the actual issue will be addressed with an additional migration hook in 3.8.
+        if (partitionService.hasOnGoingMigrationLocal() || mapServiceContext.getService().getMigrationsInFlight() > 0) {
```

Comment:
do we still need this check? you are already tracking the partition owner migrations yourself. 'hasOnGoingMigrationLocal()` returns `true` for both partition owner and backup migrations. If current node is master, it also returns `true` for pending migrations. So it can cause a long delay since index will be disabled for irrelevant partitions.